lib/pages/tag_page.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
+import 'package:qiita_client_yukiy/models/tag.dart';
+import 'package:qiita_client_yukiy/services/qiita_client.dart';
+import 'package:qiita_client_yukiy/ui_components/tag_grid_view.dart';
+import 'package:qiita_client_yukiy/ui_components/upper_bar.dart';
+
+class TagPage extends StatefulWidget {
+  TagPage({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  State<TagPage> createState() => _TagPageState();
+}
+
+class _TagPageState extends State<TagPage> {
+  List<Tag> listTags = [];
+  late bool _isLoading = true;
+  int pageNumber = 1;
+  late Future<List<Tag>> futureTags;
+
+  @override
+  void initState() {
+    _fetchTagData();
+    super.initState();
+  }
+
+  Future<void> _fetchTagData() async {
+    _isLoading = false;
+    futureTags = QiitaClient.fetchTags(pageNumber);
+    listTags.addAll(await futureTags);
```

Comment:
awaitをメソッドの引数に入れるのは余り好ましく無いです！
実際にどのメソッドがawaitされているのかパッと見分からないからです。
以下のように変更した方がいいと思います！
```suggestion
    futureTags = await QiitaClient.fetchTags(pageNumber);
    listTags.addAll(futureTags);
```

Reply:
コメントありがとうございます！
以下変更してみたのですが、画像のようなエラーが出てしまいました。試しに"Add cast"をしてみたところ、エラーは消えましたが、画面にタグが表示されなくなってしまいました。
<img width="900"  src="https://user-images.githubusercontent.com/102783724/222935489-a5fbf980-53f1-4221-96ae-6d248a25cbea.png">

<img width="911" alt="スクリーンショット 2023-03-05 9 45 03" src="https://user-images.githubusercontent.com/102783724/222935492-016317dc-742d-4a8e-8d31-9d5372b2fa71.png">

### Add cast後
<img width="300" alt="スクリーンショット 2023-03-05 9 45 03" src="https://user-images.githubusercontent.com/102783724/222935493-2d11ea32-799a-4602-898b-18dd9d217c48.png">

Reply:
BootCamp内でさらに修正したので記録します！
今回のエラーは修正前のfutureTagsと、awaitをつけたことによってfutureが消えたfetchTagsの型が違うことによって起きたエラーだと理解しました。そこで新たにList<Tag>のnewFetchTagという変数を作成し、futureTagsと入れ替えることで解決できました。以下画像です。

### 修正前
<img width="911" alt="スクリーンショット 2023-03-05 9 45 03" src="https://user-images.githubusercontent.com/102783724/222935492-016317dc-742d-4a8e-8d31-9d5372b2fa71.png">

### 修正後
<img width="586" alt="スクリーンショット 2023-03-05 10 26 44" src="https://user-images.githubusercontent.com/102783724/222936608-7eccd1fe-b48f-4e90-9225-ad927eb781ee.png">

Reply:
futureTagsがFuture型だったのを忘れていました！申し訳ありません。
修正後のコードに関してはOKです！